design/ceph/admission-controller.md

```diff
@@ -0,0 +1,182 @@
+# Admission Controllers in Rook
+
+# Background
+
+This proposal is to add support for admission controllers in Rook. An admission controller is a piece of code that intercepts requests to the Kubernetes API server prior to persistence of the object, but after the request is authenticated and authorized
+
+ There are two special controllers: MutatingAdmissionWebhook and ValidatingAdmissionWebhook.
+ 
+ Mutating controllers may modify the objects they admit but validation controllers are only allowed to validate requests.
+
+Currently, user can manipulate CRD specs with any values which may result in Rook not functioning as expected. The present validation method in Kubernetes is the OpenAPIValidation schema validation which can be used for basic validations like checking type of data, providing a range for the values etc but anything more complex (checking resource availability, network status, error handling) would not be possible under this scenario.
+ 
+# Admission Controllers Concept
+
+![AC Image](./design-pic.png)
```

Comment:
nit: how about adding the `ac-` prefix to the png filename?

Reply:
Fixed